src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -438,21 +440,61 @@ private async Task<Project> UpdateProjectInfoAsync(Project project, Checksum inf
                 Func<Solution, ImmutableArray<DocumentId>, Solution> removeDocuments,
                 CancellationToken cancellationToken) where TDocumentState : TextDocumentState
             {
-                using var _1 = PooledHashSet<Checksum>.GetInstance(out var olds);
-                using var _2 = PooledHashSet<Checksum>.GetInstance(out var news);
+                using var _1 = PooledDictionary<DocumentId, Checksum>.GetInstance(out var oldDocumentIdToChecksum);
+                using var _2 = PooledDictionary<DocumentId, Checksum>.GetInstance(out var newDocumentIdToChecksum);
```

Comment:
aligned with projects, and cleaner as dictionaries for the processing we need to do (we want the doc-ids, and htis preserves tehm).